src/spec-configuration/containerCollectionsOCI.ts

```diff
@@ -124,20 +131,49 @@ export function getRef(output: Log, input: string): OCIRef | undefined {
 	input = input.toLowerCase();
 
 	const indexOfLastColon = input.lastIndexOf(':');
+	const indexOfLastAtCharacter = input.lastIndexOf('@');
 
 	let resource = '';
-	let version = ''; // TODO: Support parsing out manifest digest (...@sha256:...)
-
-	// 'If' condition is true in the following cases:
-	//  1. The final colon is before the first slash (a port) :  eg:   ghcr.io:8081/codspace/features/ruby
-	//  2. There is no version :      				   			 eg:   ghcr.io/codspace/features/ruby
-	// In both cases, assume 'latest' tag.
-	if (indexOfLastColon === -1 || indexOfLastColon < input.indexOf('/')) {
-		resource = input;
-		version = 'latest';
+	let tag: string | undefined = undefined;
+	let digest: string | undefined = undefined;
+
+	if (indexOfLastAtCharacter !== -1) {
+		// The version is specified by digest
+		// eg: ghcr.io/codspace/features/ruby@sha256:abcdefgh
+		resource = input.substring(0, indexOfLastAtCharacter);
+		const digestWithHashingAlgorithm = input.substring(indexOfLastAtCharacter + 1);
+		const splitOnColon = digestWithHashingAlgorithm.split(':');
+		if (splitOnColon.length !== 2) {
+			output.write(`Failed to parse digest '${digestWithHashingAlgorithm}'.   Expected format: 'sha256:abcdefghijk'`, LogLevel.Error);
+			return undefined;
+		}
+
+		if (splitOnColon[0] !== 'sha256') {
+			output.write(`Digest algorithm for input '${input}' failed validation.  Expected hashing algorithm to be 'sha256'.`, LogLevel.Error);
```

Comment:
Do we rely on it being sha256?

Reply:
Yes, there is one section where we compute a sha256 of a manifest to generate the digest.  That's currently hardcoded.  I don't believe changing the hashing algorithm changing is anything we need to worry about for a while - I think it's likely appended here in case sha256 is found to be broken in the future and a new hashing algortithm needs to be swapped out for it.